front50-core/src/test/java/com/netflix/spinnaker/front50/pipeline/DefaultPipelineDAOTest.java

```diff
@@ -0,0 +1,118 @@
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.front50.pipeline;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.Mockito.*;
+
+import com.netflix.spinnaker.front50.api.model.pipeline.Pipeline;
+import com.netflix.spinnaker.front50.model.pipeline.DefaultPipelineDAO;
+import com.netflix.spinnaker.kork.sql.test.SqlTestUtil;
+import java.util.Collection;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.Mockito;
+
+public abstract class DefaultPipelineDAOTest extends PipelineDAOSpec<DefaultPipelineDAO> {
+
+  protected DefaultPipelineDAO pipelineDAO;
+
+  @Override
+  public DefaultPipelineDAO getInstance() {
+    return getDefaultPipelineDAO();
+  }
+
+  public abstract DefaultPipelineDAO getDefaultPipelineDAO();
+
+  @BeforeEach
+  public void setup() {
+    this.pipelineDAO = Mockito.spy(getDefaultPipelineDAO());
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "'app', 'pipelineNameA', 'NameA', 'pipelineNameA'",
+    "'app', 'pipelineNameA', , 'pipelineNameA'",
+    "'app', , , "
+  })
+  public void shouldReturnCorrectPipelinesWhenRequestingPipelinesByApplicationWithNameFilter(
+      String applicationName,
+      String pipelineName,
+      String pipelineNameFilter,
+      String expectedPipelineName) {
+
+    Pipeline pipeline = new Pipeline();
+    pipeline.setId("0");
+    pipeline.setApplication(applicationName);
+    pipeline.setName(pipelineName);
+
+    doReturn(List.of(pipeline)).when(pipelineDAO).all(anyBoolean());
+
+    Collection<Pipeline> pipelines =
+        pipelineDAO.getPipelinesByApplication("app", pipelineNameFilter, true);
+
+    Pipeline resultPipeline = pipelines.iterator().next();
+    assertEquals(resultPipeline.getName(), expectedPipelineName);
+    assertEquals(resultPipeline.getApplication(), "app");
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "'app', , 'NameA'",
+    "'bad', 'pipelineNameA', 'NameA'",
+    "'bad', , 'NameA'",
+    "'bad', 'pipelineNameA', ",
+    "'bad', , "
+  })
+  public void shouldReturnNoPipelinesWhenRequestingPipelinesByApplicationWithNameFilter(
+      String applicationName, String pipelineName, String pipelineNameFilter) {
+
+    Pipeline pipeline = new Pipeline();
+    pipeline.setId("0");
+    pipeline.setApplication(applicationName);
+    pipeline.setName(pipelineName);
+
+    doReturn(List.of(pipeline)).when(pipelineDAO).all(true);
+
+    Collection<Pipeline> pipelines =
+        pipelineDAO.getPipelinesByApplication("app", pipelineNameFilter, true);
+
+    assertEquals(0, pipelines.size());
+  }
+}
+
+class SqlDefaultPipelineDAOTest extends DefaultPipelineDAOTest {
+
+  private SqlTestUtil.TestDatabase database = SqlTestUtil.initTcMysqlDatabase();
```

Comment:
The groovy code had
```
  @AutoCleanup("close")
```
so I think we need to call database.close(), I guess after calling
```
SqlTestUtil.cleanupDb(database.context);
```
in the `cleanup()` method.

Reply:
added and tested.